metabase/metabase
# Collections changes feature branch

This is the feature branch for the work in #19758.

The primary pieces of this feature branch have been reviewed in separate PRs, with a few exceptions for a handful of small PRs at the end that I will specify.

- Add `PinnedItemCard`: https://github.com/metabase/metabase/pull/19765
- Add `PinnedItemOverview` and replace the existing pinned items table: https://github.com/metabase/metabase/pull/19787
- Miscellaneous UI changes to collections page: https://github.com/metabase/metabase/pull/19792
- Show card visualizations in pinned overview: https://github.com/metabase/metabase/pull/19813
- Make pinned cards draggable and droppable into the collections sidebar: https://github.com/metabase/metabase/pull/19829
- (**unreviewed**) unit & e2e tests for the new pinned overview section and ui changes: https://github.com/metabase/metabase/pull/19851
- (**unreviewed**) a changed empty state: https://github.com/metabase/metabase/pull/19857
- (**unreviewed**) some additional styling tweaks from @mazameli: https://github.com/metabase/metabase/pull/19853

**Testing**
- create a new collection. The empty state should show a new banner informing the user that they can pin items. 
- add questions and dashboards and models to the collection. The banner should persist until the user has pinned an item.
- pin a dashboard. A card for the dashboard should show up. The card should be draggable to the left sidebar to move it to a different folder. Clicking the card should take the user to the dashboard. Hovering over the card should show an action menu in the top right corner of the card. Test all the various actions, unpinning, duplicating, archiving, and moving. 
- pin a model. It should look very similar to the dashboard card except that it is separated by some label text. Should work the same. Clicking should take user to the model in the query builder. 
- test long descriptions and names for dashboards and models
- Pin a question. The visualization for the question should show sorta like if we were on a dashboard. The entire viz should be clickable to take the user to the saved question. An action menu should show in the top right corner like the dashboard and model cards. Dragging the visualization should show a card for performance reasons 
- the above items should be ordered alphabetically
